### DISCUSSION TITLE
Display warnings on projects with known security vulnerabilities

### DISCUSSION THREAD

**Discussion by Prospector** [modrinth/.github#171] [Feature requests]
For example, MC Mad Gadget (aka Bleeding Pipe) https://unascribed.com/mc-mad-gadget.html

Mods like ComputerCraft should remain for archival purposes but have a security notice put on it that it may be dangerous to use on public Minecraft servers. The notice should be specific to the mod and vulnerability in question to tell them what exactly the danger is. For example, ComputerCraft is still fine to use in Single player or trusted private Minecraft servers.

**Comment by Madis0** [comment]
* The warning should primarily be shown per file, similar to "alpha" and "beta" markings. 
* Project-level notification should only be shown heuristically, to avoid scaring the users and inducing warning fatigue. Heuristics may include: the only file is unsafe, the latest file is unsafe, the latest file per MC version is unsafe, etc.
* Ideally, the files could also be filtered out on the list by default.
* On download click, an interstitial should explain the problem and prompt the user whether they want to continue.
* If you have the statistics to determine that, and the user has an account, notify them about recent downloads being found as insecure.
* Modpack versions containing insecure mods should apply the same logic per file and per heuristics.
* Notify the mod authors when detected, advise them to update mod and delete/archive the file.
